### ISSUE TITLE
False positive reportUnnecessaryIsInstance on dicts

### ISSUE THREAD

**Issue by lovetox** [microsoft/pyright#7827]
**Describe the bug**

```python

from typing import Any


class SomeClass:
    def __init__(self) -> None:
        self._data: dict[str, Any] = {}

    def set_data(self, data: dict[str, Any]):
        if not isinstance(data, dict):
            raise ValueError
        self._data = data

    def load(self) -> dict[str, Any]:
        return self._data

```
this raises now 

`Unnecessary isinstance call; "dict[str, Any]" is always an instance of "dict[Unknown, Unknown]" `

This seems a common pattern in libraries, i offer some method and want to do a validation in set_data(), its not unnecessary to call isinstance here.

Further what makes me think this is a bug is `dict[Unknown, Unknown]`

pyright: 1.1.361

**Comment by erictraut** [comment]
Pyright is working as intended here. This isn't a false positive. From a typing perspective, this `isinstance` check is unnecessary, and this is what `reportUnnecessaryIsInstance` is designed to detect. 

This check is a strict-mode check that is off by default. If you don't want to use it in your code, you can simply leave it disabled.
